### PR TITLE
fix:assigned projectId from auth if projectId is not set

### PR DESCRIPTION
**Pull request by laljikanjareeya**
in the function `keyToLegacyUrlsafe` if user do not pass projectId in option or not set environment variable(DATASTORE_PROJECT_ID) than library is not replacing {{projectId}} with default environment projectId as we don't have an external call here.

**Solution :**
Assign `projectId` if it's default i.e. {{projectId}}.
Here function `keyToLegacyUrlsafe` has been modified with `async` because `auth.getProjectId()` returns promise, it's breaking changes.

### CODE REVIEW

**Comment by AVaksman** [comment]
Please make it a callback function.

**Comment by AVaksman** [comment]
And if we want to support `callback` and `promise` looks like we will need `promisifyAll` here.

File: src/index.ts [hunk 1]
```diff
@@ -760,10 +778,37 @@ class Datastore extends DatastoreRequest {
...
+   *   }
+   *   console.log(urlSafekey);
+   *
+   * @example
+   * const {Datastore} = require('@google-cloud/datastore');
+   * const datastore = new Datastore();
+   * const key = datastore.key(['Task', 123]);
+   * const locationPrefix = 's~';
+   *
+   * const urlSafekey = await datastore.keyToLegacyUrlsafe(key, locationPrefix);
```
Comment (AVaksman) [inline src/index.ts:796]: Just add a doc line above here
```
* If the callback is omitted, we'll return a Promise.
``` 

Similar to what they use in other places

Comment (laljikanjareeya) [reply src/index.ts:796]: Done!

**Pull request closed without merging**